### ISSUE TITLE
[Additional Service] Add DHL cash on delivery

### ISSUE THREAD

**Issue by andrecedik** [awsmug/shipcloud-for-woocommerce#68]
Cash on delivery in Germany is still one of the main ways to pay for online orders. Thus integrating this *additional service* is key to reaching a larger buyer audience. 

shipcloud currently supports [cash on delivery for DHL](https://developers.shipcloud.io/recipes/#dhl---cash-on-delivery). To create a shipment that gets payed by c.o.d. the store admin will have to configure bank account data that should be used for transferring the money to the seller.

Example:
```json
"additional_services": [
    {
      "name": "cash_on_delivery",
      "properties": {
        "amount": 123.45,
        "currency": "EUR",
        "bank_account_holder": "Max Mustermann",
        "bank_name": "Musterbank",
        "bank_account_number": "DE12500105170648489890",
        "bank_code": "BENEDEPPYYY",
        "reference1": "reason for transfer"
      }
    }
  ]
```

As you can see shipcloud needs the combination of IBAN and SWIFT code as well as the name of the account holder and the bank and of course the amount that should be taken from the buyer. You can also specify the parameter `reference1` which will be used for identifying the correct transaction on your account statement.

**Comment by ScreamingDev** [comment]
@andrecedik I have questions:

- Does Shipcloud or third party validate the bank information?
- Do we show an error when such information is not given in the backend?

**Comment by andrecedik** [comment]
shipcloud does not validate nor save the data in any way for security reasons. We're simply using it to make the request at the carrier. When the data is missing and you're trying to create a c.o.d shipment you will get an error message by the carrier.

**Comment by andrecedik** [comment]
Fixed via #152
